File: OOP/Week 07/Task/cd.py
from item import Item

class CD(Item):

    def __init__(self, title, artist, playing_time, number_of_tracks):
        self.artist = artist
        self.number_of_tracks = number_of_tracks
        super().__init__(title, playing_time)

    def get_artist(self):
        return self.artist

    def set_artist(self, artist):
        self.artist = artist

    def get_number_of_tracks(self):
        return self.number_of_tracks

    def set_owned(self, owned):
        super().set_owned(owned)

    def get_owned(self):
        return super().get_owned()

    def info(self):
        print("Artist:", self.artist)
        super().info()
        print("Number of tracks:", self.number_of_tracks)

cd = CD("title", "Artist", 100, 12)
cd.info()

File: OOP/Java OOP/Week 09/Task02/CD.java
package Task02;

public class CD extends Item {
    private String artist;
    private int numberOfTracks;

    CD(String title, String artist, String playingTime, int numberOfTracks) {
        super(title, playingTime);

        this.artist = artist;
        this.numberOfTracks = numberOfTracks;
    }

    public String getArtist() {
        return this.artist;
    }

    public void setArtist(String artist) {
        this.artist = artist;
    }

    public int getNumberOfTracks() {
        return this.numberOfTracks;
    }

    public void setNumberOfTracks(int numberOfTracks) {
        this.numberOfTracks = numberOfTracks;
    }

    public boolean getOwned() {
        return super.getOwned();
    }

    public void setOwned(boolean owned) {
        super.setOwned(owned);
    }

    public void info() {
        System.out.println("Artist: " + this.artist);
        super.info();
        System.out.println("Number of tracks: " + this.numberOfTracks);
    }
    
}
